src/labone/mock/hpk_functionality.py

```diff
@@ -0,0 +1,67 @@
+from abc import ABC, abstractmethod
+from contextlib import contextmanager
+import json
+
+from labone.core import ListNodesFlags, ListNodesInfoFlags
+from labone.core.helper import LabOneNodePath
+from labone.core.session import NodeInfo
+from labone.core.subscription import StreamingHandle
+from labone.core.value import AnnotatedValue
+from labone.mock.mock_server import ServerTemplate
+
+
+class HpkMockFunctionality(ABC):
```

Comment:
nit-pick: rename to `SessionMockFunctionality` also I would move it into `session_mock_template`